lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java

```diff
@@ -57,41 +57,33 @@
                 query = "SELECT a FROM SocActivity a WHERE a.hidden = false  AND a.updatedDate > :sinceTime AND exists (SELECT item FROM a.streamItems item WHERE item.ownerId in (:owner)) ORDER BY a.updatedDate ASC"),
         @NamedQuery(name = "SocActivity.getActivityByOwnerAndProviderId",
                 query = "SELECT a FROM SocActivity a WHERE a.hidden = false AND a.providerId = :providerId AND exists (SELECT item FROM a.streamItems item WHERE item.ownerId in (:owner)) ORDER BY a.updatedDate DESC"),
+        @NamedQuery(name = "SocActivity.getConnections",
+                query = "SELECT conn1.receiver.id from SocConnection conn1 where conn1.status = :connStatus  and conn1.receiver.id = :identityId" +
+                        " union " +
+                        "SELECT conn2.sender.id from SocConnection conn2 where conn2.status = :connStatus and conn2.sender.id = :identityId "),
```

Comment:
Looks better to move to ConnectionEntity, yes.
I don't have a problem to use "union" using Hibernate (I don't think we will want to transparently migrate to another JPA impl). If this is not convincing you, I can switch to use Standard SQL, WDYT ?